src/Platform_Core/org/lobobrowser/context/AlgorithmDigest.java

```diff
@@ -0,0 +1,131 @@
+package org.lobobrowser.context;
+
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AlgorithmDigest implements Comparable<AlgorithmDigest> {
+
+  private final String algorithm;
+  private final String digest;
+  private final Integer strength;
+  private final static Map<String, Integer> str = createMap();
+
+  public AlgorithmDigest(String algorithm, String digest, Integer strength) {
+    super();
+    this.algorithm = algorithm;
+    this.digest = digest;
+    this.strength = strength;
+  }
+
+  public static boolean validate(final byte[] input, final String integrity) {
+
+    final List<AlgorithmDigest> algDigests = parseMetadata(integrity);
+    System.out.println("Strongest algorithms: " + algDigests);
+
+    System.out.println("The value of the integrity attribute is: " + integrity);
+
+    if (algDigests == null)
+      return true;
+
+    final boolean matchFound = algDigests.stream().anyMatch((algDigest) -> {
+      final String encodedResult = algDigest.getHash(input);
+      return encodedResult.equals(algDigest.digest);
+    });
+
+    return matchFound;
+
+  }
+
+  public static List<AlgorithmDigest> parseMetadata(final String integrity) {
+
+    if (integrity == null || integrity.length() == 0) {
+      return null;
+    }
+
+    System.out.println(integrity);
+    final String[] tokens = integrity.split("\\s+");
+
+    final List<AlgorithmDigest> hashes = getHashes(tokens);
+
+    if (hashes.isEmpty())
+      return null;
+
+    return strongestAlgDigests(hashes);
+
+  }
+
+  private static List<AlgorithmDigest> getHashes(final String[] tokens) {
+    final List<AlgorithmDigest> hashes = new ArrayList<AlgorithmDigest>();
+
+    for (int i = 0; i < tokens.length; i++) {
```

Comment:
Try converting to enhanced `for` loop. In eclipse `Ctrl 1` should suggest it automatically.